src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -252,54 +253,43 @@ await pipeline.ExecuteAsync(async (ctx) =>
 
         static async Task AcceptRedisInsightEula(ILogger resourceLogger, HttpClient client, CancellationToken ct)
         {
-            using (var stream = new MemoryStream())
-            {
-                using var writer = new Utf8JsonWriter(stream);
 
-                writer.WriteStartObject();
+            var settings = new
+            {
+                agreements = new
+                {
+                    eula = true,
+                    analytics = false,
+                    notifications = false,
+                    encryption = false
+                }
+            };
 
-                writer.WritePropertyName("agreements");
-                writer.WriteStartObject();
-                writer.WriteBoolean("eula", true);
-                writer.WriteBoolean("analytics", false);
-                writer.WriteBoolean("notifications", false);
-                writer.WriteBoolean("encryption", false);
-                writer.WriteEndObject();
+            var content = JsonContent.Create(settings);
 
-                writer.WriteEndObject();
+            var apiUrl = $"/api/settings";
 
-                await writer.FlushAsync(ct).ConfigureAwait(false);
-                stream.Seek(0, SeekOrigin.Begin);
-                string json = Encoding.UTF8.GetString(stream.ToArray());
-                var content = new StringContent(json, Encoding.UTF8, "application/json");
-
-                var apiUrl = $"/api/settings";
-
-                var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
-                {
-                    Delay = TimeSpan.FromSeconds(2),
-                    MaxRetryAttempts = 5,
-                }).Build();
+            var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
+            {
+                Delay = TimeSpan.FromSeconds(2),
+                MaxRetryAttempts = 5,
+            }).Build();
```

Comment:
Isn't there a way to only retry on transient http failures? This will retry on a 404 no?